Tasks/VsTest/vstest.ts

```diff
@@ -105,6 +107,63 @@ function getTestAssemblies(): Set<string> {
     return new Set(testAssemblyFiles);
 }
 
+function getVstestArgsArray(settingsFile: string): string[] {
+    var argsArray: string[] = [];
+    testAssemblyFiles.forEach(function (testAssembly) {
+        var testAssemblyPath = testAssembly;
+        //To maintain parity with the behaviour when test assembly was filepath, try to expand it relative to build sources directory.
+        if (sourcesDirectory && !pathExistsAsFile(testAssembly)) {
+            var expandedPath = path.join(sourcesDirectory, testAssembly);
+            if (pathExistsAsFile(expandedPath)) {
+                testAssemblyPath = expandedPath;
+            }
+        }
+        argsArray.push(testAssemblyPath);
+    });
+    if (testFiltercriteria) {
+        argsArray.push("/TestCaseFilter:" + testFiltercriteria);
+    }
+    if (settingsFile && pathExistsAsFile(settingsFile)) {
+        argsArray.push("/Settings:" + settingsFile);
+    }
+    if (codeCoverageEnabled) {
+        argsArray.push("/EnableCodeCoverage");
+    }
+    if (otherConsoleOptions) {
+        argsArray.push(otherConsoleOptions);
+    }
+    argsArray.push("/logger:trx");
+    if (pathtoCustomTestAdapters) {
+        if (pathExistsAsDirectory(pathtoCustomTestAdapters)) {
+            argsArray.push("/TestAdapterPath:\"" + pathtoCustomTestAdapters + "\"");
+        }
+        else {
+            argsArray.push("/TestAdapterPath:\"" + path.dirname(pathtoCustomTestAdapters) + "\"");
+        }
+    }
+    else if (sourcesDirectory && isNugetRestoredAdapterPresent(sourcesDirectory)) {
+        argsArray.push("/TestAdapterPath:\"" + sourcesDirectory + "\"");
+    }
+    return argsArray;
+}
+
+function addVstestArgs(argsArray: string[], vstest: any) {
+    argsArray.forEach(function (arr) {
+        vstest.arg(arr);
+    });
+}
+
+function updateResponsefile(argsArray: string[], responseFile: string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    fs.appendFile(responseFile, os.EOL + argsArray.join(os.EOL), function (err) {
+        if (err) {
+            defer.reject(err);
+        }
+        defer.resolve("\"" + responseFile + "\"");
```

Comment:
For wrapping the filename under quotes("), to avoid error in cases when filename contains spaces.